README.md

```diff
@@ -4,6 +4,8 @@ The **[Amazon DynamoDB][ddb] Client-side Encryption in Java** supports encryptio
 
 A typical use of this library is when you are using [DynamoDBMapper][ddbmapper], where transparent protection of all objects serialized through the mapper can be enabled via configuring an [AttributeEncryptor][attrencryptor].
 
+**Important: Use `SaveBehavior.CLOBBER` with `AttributeEncryptor`. If you do not do so you risk corrupting your signatures and encrypted data.**
```

Comment:
Explain why this is the case?

Reply:
**Important: Use `SaveBehavior.CLOBBER` with `AttributeEncryptor`. If you do not do so you risk corrupting your signatures and encrypted data.** The default behavior of the mapper is to leave unknown fields unchanged; when running a fleet with mixed versions of your record classes, this can result in fields being left in a record that did not factor into the signature computation, which will fail to decrypt. By specifying `CLOBBER`, the entire record is replaced at once, removing unknown fields, and ensuring that the signature is correct.

Reply:
+1 - I want to learn more about why this is the desired behavior

Using CLOBBER basically disables the Optimistic Locking which is a super useful feature provided by DynamoDB. This says we shouldn't use AttributeEncryptor and Optimistic Locking together. 

We support partial encryption and signing today, can those fields be updated safely without CLOBBER save?

Reply:
We will re-visit how optimistic logging works with encrypted data. It definitely will not work without a version field.  This change is still important due to the need to warn people about the risk of data-corruption.